Allow drag-to-scroll

tamo opened the pull request:
Convert mouse-motion events to wheel-scroll events
while the button is held down.

Without this modification, this utility works BETWEEN two clicks.
It is NOT drag, despite the name.

tamo committed:
Allow drag-to-scroll

Convert mouse-motion events to wheel-scroll events
while the button is held down.

Without this modification, this utility works BETWEEN two clicks.
It is NOT drag, despite the name.

sn1c commented:
I think this is what I was looking for when stumbling upon DragScroll. How can I compile this on my machine?

tamo commented:
> I think this is what I was looking for when stumbling upon DragScroll. How can I compile this on my machine?

Open this repository with Xcode as usual.
https://developer.apple.com/documentation/xcode/building-and-running-an-app

tamo commented:
Or simply run `xcodebuild clean build analyze -scheme DragScroll -project DragScroll.xcodeproj`

Also you can set up [GitHub Actions](https://github.com/actions/starter-workflows/blob/main/ci/objective-c-xcode.yml) like [this](https://github.com/tamo/drag-scroll/blob/github-actions/.github/workflows/objective-c-xcode.yml)

sn1c commented:
I managed to compile it, thanks a lot! 

~~Unfortunately, I'm still facing some issues regarding permission. Even after I allow it in the accessibility settings, it still asks for it everytime and does not work. I once got it work by restarting it a lot and having it running multiple times, but it doesn't work consistently. Did you face a similar issue before?~~

Nevermind, I managed to fix it using 
`tccutil reset Accessibility com.emreyolcu.DragScroll`

Thanks a lot! It works like a charm!

sn1c commented:
In fact, I'd like to even disable the toggling and only have the drag. Would be nice to have this as a feature, possibly so that it can be enabled/disabled. If you are interested and can give me some pointers, I can take a look.

tamo committed:
Add option to disable toggle behavior

Run `defaults write com.emreyolcu.DragScroll toggle -int 0`
if you don't want the mouse button to toggle scrolling
